utilities/transactions/write_unprepared_txn.cc

```diff
@@ -46,7 +46,408 @@ SequenceNumber WriteUnpreparedTxnReadCallback::MaxUnpreparedSequenceNumber() {
 WriteUnpreparedTxn::WriteUnpreparedTxn(WriteUnpreparedTxnDB* txn_db,
                                        const WriteOptions& write_options,
                                        const TransactionOptions& txn_options)
-    : WritePreparedTxn(txn_db, write_options, txn_options), wupt_db_(txn_db) {}
+    : WritePreparedTxn(txn_db, write_options, txn_options), wupt_db_(txn_db) {
+  max_write_batch_size_ = txn_options.max_write_batch_size;
+  write_batch_.SetMaxBytes(0);
+}
+
+WriteUnpreparedTxn::~WriteUnpreparedTxn() {
+  if (!unprep_seqs_.empty()) {
+    assert(log_number_ > 0);
+    assert(id_ > 0);
+    assert(name_.size() > 0);
+
+    // We should really always rollback, but some tests rely on the destructor
+    // not doing the rollback for us for crash testing.
```

Comment:
not doing.

I want to prevent the destructor from rolling back, so that I can test that recovery does the rollback (not the destruction of transaction).

Reply:
Let me try to reword, since this might be too confusing.